pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -458,6 +458,14 @@ def test_constructor_with_categorical_categories(self):
         result = Categorical(["a", "b"], categories=CategoricalIndex(["a", "b", "c"]))
         tm.assert_categorical_equal(result, expected)
 
+    def test_construction_with_null(self, nulls_fixture):
+        # https://github.com/pandas-dev/pandas/issues/31927
+        values = ["a", nulls_fixture, "b"]
+        result = Categorical(np.array(values, dtype=object))
+        expected = Categorical(values)
```

Comment:
I am not sure this is a very good test. I mean: it is testing that lists vs object array are giving the same result (which is useful anyhow, as those should be consistent), but it is not testing *how* they are now constructed (eg it won't "preserve" pd.NA, and this is also not tested)

Reply:
@dsaxton can you parameterize this on klass (np.array and list), then hard code the results in a categorical (meaning use _from_codes and an explict list of categories)